src/related.tsx

```diff
@@ -120,10 +120,10 @@ class Related extends KalturaPlayer.core.BasePlugin {
     const preplayBackPlayOverlayProps = {
       relatedManager,
       onLoaded: (callback: (nextEntries: []) => void) => {
-        relatedManager.listen(RelatedEvent.HIDDEN_STATE_CHANGED, ({payload}: {payload: []}) => callback(payload));
+        relatedManager.listen(RelatedInternalEvent.HIDDEN_STATE_CHANGED, ({payload}: {payload: []}) => callback(payload));
```

Comment:
Please use the event manager decorator and call removeAll on onUnmount hook

Don't use the same event manager of the plugin

Reply:
as we discussed, fixing it on the other PR https://github.com/kaltura/playkit-js-related/pull/33